docs/explanation/workflow.md

```diff
@@ -0,0 +1,8 @@
+# The charm's workflow
+1. When the charm starts, it goes into a blocked state until TLS is configured, either via a config option or via an integration with a TLS provider charm.
+
+2. Once TLS is configured, a gateway object containing a http and a https listener is created: The minimal gateway object. The charm will remain in blocked state until integration with an ingress requirer charm is established.
```

Comment:
I think it might be more readable to restructure the first sentence as "Once TLS is configured, a minimal gateway object containing a http and https listener is created." Unless splitting "minimal gateway object" from the rest of the sentence serves a specific purpose?

Nit-pick: The charm will remain in a block state...

Reply:
Fixed and added a diagram